feebleline.el

```diff
@@ -72,47 +85,41 @@
   (if feebleline-mode
       (progn
         (setq feebleline/mode-line-format-default mode-line-format)
-        (setq feebleline/timer (run-with-timer 0 0.1 'mode-line-proxy-fn))
+        (setq feebleline/timer (run-with-timer 0 0.1 'feebleline-mode-line-proxy-fn))
         (custom-set-variables '(mode-line-format nil))
         (ad-activate 'handle-switch-frame)
-        (add-hook 'focus-in-hook 'mode-line-proxy-fn))
+        (add-hook 'focus-in-hook 'feebleline-mode-line-proxy-fn))
     (custom-set-variables
      '(mode-line-format feebleline/mode-line-format-default))
     (cancel-timer feebleline/timer)
     (ad-deactivate 'handle-switch-frame)
-    (remove-hook 'focus-in-hook 'mode-line-proxy-fn)))
-
-;; (defvar msg-string)
-;; (defun mode-line-proxy-string-fn () (interactive)
-;;   "Hej."
-;;   (setq msg-string (concat "a" "b"))
-;;   (message "%s" msg-string))
-
-(defun message-buffer-file-name-or-nothing ()
+    (remove-hook 'focus-in-hook 'feebleline-mode-line-proxy-fn)))
+
+(defun feebleline--mode-line-part (part)
+  "Return a PART (an element) of `feebleline-mode-line-text` as a propertized string."
+  (let ((text (apply #'format (append (list (car part))
+                                      (mapcar #'eval (cadr part)))))
+        (props (elt part 2)))
+    (when props
+      (add-text-properties 0 (length text) props text))
+    text))
+
+(defun feebleline-message-buffer-file-name-or-nothing ()
   "Replace echo-area message with mode-line proxy."
-  (setq feebleline/default-log-max message-log-max)
-  (unwind-protect
-      (progn
-        (setq message-log-max nil)
-        (if buffer-file-name
-            (message "[%s] %s/%s @ %s"
-                     (format-time-string "%H:%M:%S")
-                     (string-to-number (format-mode-line "%l"))
-                     (current-column)
-                     (buffer-file-name))))
-    (setq message-log-max feebleline/default-log-max)))
-
-(defun mode-line-proxy-fn ()
+  (when buffer-file-name
+    (let ((message-log-max nil))
```

Comment:
By let-binding `message-log` we give it a scope local value. When the scope exists, it will get the previous value again. Thus the global variable you had wasn't needed.